src/keys.rs

```diff
@@ -281,76 +292,60 @@ impl IssuanceAuthorizingKey {
     }
 }
 
 /// A key used to validate issuance authorization signatures.
 ///
 /// Defined in [ZIP 227: Issuance of Zcash Shielded Assets § Issuance Key Generation][IssuanceZSA].
-/// Note that this is $\mathsf{ik}^\mathbb{P}$, which by construction is equivalent to
-/// $\mathsf{ik}$ but stored here as a RedPallas verification key.
 ///
 /// [IssuanceZSA]: https://qed-it.github.io/zips/zip-0227#issuance-key-derivation
-#[derive(Debug, Clone, PartialOrd, Ord)]
-pub struct IssuanceValidatingKey(VerificationKey<IssuanceAuth>);
+#[derive(Debug, Clone)]
+pub struct IssuanceValidatingKey(schnorr::VerifyingKey);
 
 impl From<&IssuanceAuthorizingKey> for IssuanceValidatingKey {
     fn from(isk: &IssuanceAuthorizingKey) -> Self {
-        IssuanceValidatingKey((&(conditionally_negate(isk.derive_inner()))).into())
-    }
-}
-
-impl From<&IssuanceValidatingKey> for pallas::Point {
-    fn from(issuance_validating_key: &IssuanceValidatingKey) -> pallas::Point {
-        pallas::Point::from_bytes(&(&issuance_validating_key.0).into()).unwrap()
+        let schnorr_sk = schnorr::SigningKey::from_bytes(&isk.0).unwrap();
+        IssuanceValidatingKey(*schnorr_sk.verifying_key())
     }
 }
 
 impl PartialEq for IssuanceValidatingKey {
     fn eq(&self, other: &Self) -> bool {
-        <[u8; 32]>::from(&self.0).eq(&<[u8; 32]>::from(&other.0))
+        self.to_bytes().eq(&other.to_bytes())
     }
 }
 
 impl Eq for IssuanceValidatingKey {}
 
 impl IssuanceValidatingKey {
-    /// Converts this spend validating key to its serialized form,
-    /// I2LEOSP_256(ik).
+    /// Converts this issuance validating key to its serialized form,
+    /// in big-endian order as defined in BIP 340.
     pub fn to_bytes(&self) -> [u8; 32] {
-        // This is correct because the wrapped point must have ỹ = 0, and
-        // so the point repr is the same as I2LEOSP of its x-coordinate.
-        <[u8; 32]>::from(&self.0)
+        // This provides the big-endian encoding, via the FieldBytesEncoding trait in the k256 crate.
+        U256::decode_field_bytes(&self.0.to_bytes()).to_be_bytes()
     }
 
-    /// Constructs an Orchard issuance validating key from uniformly-random bytes.
+    /// Constructs an Orchard issuance validating key from the provided bytes.
+    /// The bytes are assumed to be encoded in big-endian order.
     ///
     /// Returns `None` if the bytes do not correspond to a valid key.
     pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
-        <[u8; 32]>::try_from(bytes)
+        schnorr::VerifyingKey::from_bytes(bytes)
             .ok()
-            .and_then(check_structural_validity)
             .map(IssuanceValidatingKey)
     }
 
     /// Verifies a purported `signature` over `msg` made by this verification key.
     pub fn verify(
         &self,
         msg: &[u8],
-        signature: &redpallas::Signature<IssuanceAuth>,
-    ) -> Result<(), reddsa::Error> {
+        signature: &schnorr::Signature,
+    ) -> Result<(), k256::schnorr::Error> {
```

Comment:
as discussed, remove the error type if the underlying function does not return `Error`

Reply:
I checked this, and the `Result<()>` that we saw in the function signature is actually just sugar to avoid writing the entire `Result<(), k256::schnorr::Error>` term.

We have the following code [linked here](https://docs.rs/signature/2.1.0/src/signature/error.rs.html#11).

```
/// Result type.
///
/// A result with the `signature` crate's [`Error`] type.
pub type Result<T> = core::result::Result<T, Error>;
```

So we would need to specify the Error type, and I think it's better this way, and there is no need to define `pub type Result<T> = ...` for us.